library/src/main/java/com/afollestad/date/controllers/DatePickerController.kt

```diff
@@ -106,7 +106,7 @@ internal class DatePickerController(
     if (notifyListeners) {
       notifyListeners(oldSelected) { calendar.clone() as Calendar }
     }
-    updateCurrentMonth(calendar)
+    updateCurrentMonth(calendar.clone() as Calendar)
```

Comment:
Since `calendar.clone() as Calendar` is also done on line 107 - it'd be better to pull this out to a `val` on line 105 and use that val in both of these places

Reply:
@afollestad  Thank you for your suggestion. Initially I tried to take this in count but isn't possible to apply due the `updateCurrentMonth()` modify that unique `calendar` clone propagate same `object` instance by the `notifyListeners()`. 

```java
val newCalendar = calendar.clone() as Calendar
    if (notifyListeners) {
      notifyListeners(oldSelected) { newCalendar }
    }
    updateCurrentMonth(newCalendar)
```

Making tests to fail with the following message: 
```java
Didn't get matching new date in emission.
expected: DateSnapshot(month=6, day=4, year=1995)
but was : DateSnapshot(month=6, day=31, year=1995) -> Value modified by updateCurrentMonth
Expected :DateSnapshot(month=6, day=4, year=1995)
Actual   :DateSnapshot(month=6, day=31, year=1995) -> Value modified by updateCurrentMonth
```

The root of the problem is on the `MonthGraph` class, IMHO it never should modify the constructor param instance,  and in case that needs maybe is better to do that with internally instance